Clients/src/store/auth.store.js

```diff
@@ -0,0 +1,64 @@
+import {defineStore} from 'pinia';
+import axiosInstance from "../services/auth.api";
+
+export const useAuthStore= defineStore('auth', {
+    state: () => ({
+        user: null,
+        token: localStorage.getItem('token') || null,
+        error: null
+    }),
+    actions:{
+        async register(credentials) {
+            try {
+                const response = await axiosInstance.post('/register', credentials);
+                this.token = response.data.token;
+                this.user = response.data.user;
+                localStorage.setItem('token', this.token);
+                axiosInstance.defaults.headers.common['Authorization'] = `Bearer ${this.token}`;
+                return { success: true, user: this.user };
+            } catch (error) {
+                this.error = error.response?.data || { message: 'Registration failed' };
+                console.error('Registration failed:', error);
+                return { success: false, error: this.error };
+            }
+        },
+        async login(credentials) {
+            try {
+                const response = await axiosInstance.post('/login', credentials);
+                this.token = response.data.token;
+                this.user = response.data.user;
+                localStorage.setItem('token', this.token);
+                axiosInstance.defaults.headers.common['Authorization'] = `Bearer ${this.token}`;
+                return { success: true, user: this.user };
+            } catch (error) {
+                this.error = error.response?.data || { message: 'Login failed' };
+                return { success: false, error: this.error };
+            }
+        },
+        async logout() {
+            try {
+                await axiosInstance.post('/logout');
+                this.token = null;
+                this.user = null;
+                localStorage.removeItem('token');
+                delete axiosInstance.defaults.headers.common['Authorization'];
+                return {success: true};
+            } catch (error) {
+                this.error = error.response?.data || { message: 'Logout failed' };
+                console.error('Logout failed:', error);
+                return {success: false, error: this.error};
+            }
```

Comment:
Do you see the common pattern between these functions? all of them use almost the same repeated code except a few changes like endpoint, let's clear it up by extracting it to a new function and reuse it for all three.